[BUG]: Import of sonatypeiq_source_control fails

### Expected Behavior

The resource should have been imported. This has failed on 2 resources I have tried to import, I will attempt to do further testing as I worked around the issue by deleting the configuration using the rest API and then recreating.

If there isn't sufficient debug output I will do a PR to add that.

### Actual Behavior

The provider crashed, see output below.

### Sonatype IQ Server Version

175

### Terraform Version

1.8.5

### Affected Resource(s)

sonatypeiq_source_control

### Terraform Configuration Files

_No response_

### Steps to Reproduce

_No response_

### Debug Output

_No response_

### Panic Output

```shell
Stack trace from the terraform-provider-sonatypeiq_v0.8.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x154eb00]

goroutine 376 [running]:
terraform-provider-sonatypeiq/internal/provider.(*sourceControlResource).Read(0xc0006ed530, {0x1b03448, 0xc0006ed4a0}, {{{{0x1b077d0, 0xc0000a4ab0}, {0x18aace0, 0xc0000a45a0}}, {0x1b09128, 0xc0000a3a40}}, 0xc0003bc060, ...}, ...)
        terraform-provider-sonatypeiq/internal/provider/source_control_resource.go:205 +0x520
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0xc0001a6dc0, {0x1b03448, 0xc0006ed4a0}, 0xc0006ed560, 0xc00037d6c8)
        github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/fwserver/server_readresource.go:101 +0x64e
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadResource(0xc0001a6dc0, {0x1b03448?, 0xc0006ed380?}, 0xc00071f500)
        github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/proto6server/server_readresource.go:55 +0x3e5
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0xc0002528c0, {0x1b03448?, 0xc0006ec9f0?}, 0xc000740ae0)
        github.com/hashicorp/terraform-plugin-go@v0.22.0/tfprotov6/tf6server/server.go:776 +0x4c3
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0x19a3160?, 0xc0002528c0}, {0x1b03448, 0xc0006ec9f0}, 0xc0003b3d00, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.22.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:482 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001bb000, {0x1b03448, 0xc0006ec960}, {0x1b08140, 0xc00040c1a0}, 0xc0004e26c0, 0xc000335290, 0x21249c8, 0x0)
        google.golang.org/grpc@v1.61.1/server.go:1385 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc0001bb000, {0x1b08140, 0xc00040c1a0}, 0xc0004e26c0)
        google.golang.org/grpc@v1.61.1/server.go:1796 +0xfec
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.61.1/server.go:1029 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 36
        google.golang.org/grpc@v1.61.1/server.go:1040 +0x135

Error: The terraform-provider-sonatypeiq_v0.8.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/internal/provider/source_control_resource.go
+++ b/internal/provider/source_control_resource.go
@@ -25,6 +25,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
 	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
 	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
 	"io"
 	"strings"
 
@@ -184,12 +185,16 @@ func (r *sourceControlResource) Read(ctx context.Context, req resource.ReadReque
 	// Get refreshed Source Control from IQ
 	ownerType := strings.Trim(state.OwnerType.String(), "\"")
 	ownerId := strings.Trim(state.ID.String(), "\"")
+	tflog.Debug(ctx, fmt.Sprintf("Importing Source Control with owner_type: %s", ownerType))
+	tflog.Debug(ctx, fmt.Sprintf("Importing Source Control with owner_id: %s", ownerId))
 	sourceControl, apiResponse, err := r.client.SourceControlAPI.GetSourceControl1(ctx, ownerType, ownerId).Execute()
 	if err != nil {
 		if apiResponse.StatusCode == 404 {
+			tflog.Debug(ctx, "Remove resource from state as no longer exists")
 			resp.State.RemoveResource(ctx)
 			return
 		} else {
+			tflog.Error(ctx, "Error Reading IQ Source Control")
 			resp.Diagnostics.AddError(
 				"Error Reading IQ Source Control",
 				"Could not read Source Control for ID "+state.ID.ValueString()+": "+err.Error(),
@@ -198,14 +203,23 @@ func (r *sourceControlResource) Read(ctx context.Context, req resource.ReadReque
 		return
 	} else {
 		// Overwrite items with refreshed state
+		tflog.Debug(ctx, fmt.Sprintf("Setting imported attributes in state %s", *sourceControl.OwnerId))
 		state.ID = types.StringValue(*sourceControl.OwnerId)
 		if ownerType == "application" {
 			state.RepositoryUrl = types.StringValue(*sourceControl.RepositoryUrl)
 		}
-		state.BaseBranch = types.StringValue(*sourceControl.BaseBranch)
-		state.RemediationPullRequestsEnabled = types.BoolValue(*sourceControl.RemediationPullRequestsEnabled)
-		state.PullRequestCommentingEnabled = types.BoolValue(*sourceControl.PullRequestCommentingEnabled)
-		state.SourceControlEvaluationsEnabled = types.BoolValue(*sourceControl.SourceControlEvaluationsEnabled)
+		if sourceControl.BaseBranch != nil {
+			state.BaseBranch = types.StringValue(*sourceControl.BaseBranch)
+		}
+		if sourceControl.RemediationPullRequestsEnabled != nil {
+			state.RemediationPullRequestsEnabled = types.BoolValue(*sourceControl.RemediationPullRequestsEnabled)
+		}
+		if sourceControl.PullRequestCommentingEnabled != nil {
+			state.PullRequestCommentingEnabled = types.BoolValue(*sourceControl.PullRequestCommentingEnabled)
+		}
+		if sourceControl.SourceControlEvaluationsEnabled != nil {
+			state.SourceControlEvaluationsEnabled = types.BoolValue(*sourceControl.SourceControlEvaluationsEnabled)
+		}
 		if sourceControl.Provider != nil {
 			state.ScmProvider = types.StringValue(*sourceControl.Provider)
 		}
@@ -260,8 +274,6 @@ func (r *sourceControlResource) Update(ctx context.Context, req resource.UpdateR
 		return
 	}
 	// Set the state to fully populated data
-	resp.Diagnostics.AddWarning(
-		"what", "what"+*sourceControl.Id+"XXXXXXX"+*sourceControl.OwnerId)
 	plan.ID = types.StringValue(*sourceControl.OwnerId)
 	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
 	if resp.Diagnostics.HasError() {
@@ -310,7 +322,7 @@ func (r *sourceControlResource) ImportState(ctx context.Context, req resource.Im
 		)
 		return
 	}
-
+	tflog.Debug(ctx, fmt.Sprintf("Importing Source Control with owner_type: %s and owner_id: %s", idParts[0], idParts[1]))
 	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("owner_type"), idParts[0])...)
 	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("owner_id"), idParts[1])...)
 }

--- a/internal/provider/source_control_resource_test.go
+++ b/internal/provider/source_control_resource_test.go
@@ -25,6 +25,38 @@ import (
 	"github.com/hashicorp/terraform-plugin-testing/terraform"
 )
 
+func TestAccSourceControlApplicationResourceMinimumConfig(t *testing.T) {
+	rand := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	resourceName := "sonatypeiq_source_control.test"
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSourceControlApplicationResourceMinimumConfig(rand),
+				Check: resource.ComposeTestCheckFunc(
+					resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr(resourceName, "repository_url", "https://github.com/sonatype-nexus-community/terraform-provider-sonatypeiq.git"),
+						resource.TestCheckNoResourceAttr(resourceName, "remediation_pull_requests_enabled"),
+						resource.TestCheckNoResourceAttr(resourceName, "pull_request_commenting_enabled"),
+						resource.TestCheckNoResourceAttr(resourceName, "source_control_evaluation_enabled"),
+						resource.TestCheckNoResourceAttr(resourceName, "base_branch"),
+					),
+				),
+			},
+			{
+				ResourceName:                         resourceName,
+				ImportState:                          true,
+				ImportStateVerify:                    true,
+				ImportStateVerifyIdentifierAttribute: "owner_id",
+				ImportStateIdFunc: func(s *terraform.State) (string, error) {
+					id := s.RootModule().Resources[resourceName].Primary.Attributes["owner_id"]
+					return fmt.Sprintf("application:%s", id), nil
+				},
+			},
+		},
+	})
+}
+
 func TestAccSourceControlApplicationResource(t *testing.T) {
 	rand := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
 	firstState := "true"
@@ -38,8 +70,9 @@ func TestAccSourceControlApplicationResource(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					resource.ComposeAggregateTestCheckFunc(
 						resource.TestCheckResourceAttr(resourceName, "remediation_pull_requests_enabled", firstState),
-						resource.TestCheckResourceAttr(resourceName, "remediation_pull_requests_enabled", firstState),
-						resource.TestCheckResourceAttr(resourceName, "remediation_pull_requests_enabled", firstState),
+						resource.TestCheckResourceAttr(resourceName, "pull_request_commenting_enabled", firstState),
+						resource.TestCheckResourceAttr(resourceName, "source_control_evaluation_enabled", firstState),
+						resource.TestCheckResourceAttr(resourceName, "base_branch", "my-cool-branch"),
 					),
 				),
 			},
@@ -58,8 +91,9 @@ func TestAccSourceControlApplicationResource(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					resource.ComposeAggregateTestCheckFunc(
 						resource.TestCheckResourceAttr(resourceName, "remediation_pull_requests_enabled", secondState),
-						resource.TestCheckResourceAttr(resourceName, "remediation_pull_requests_enabled", secondState),
-						resource.TestCheckResourceAttr(resourceName, "remediation_pull_requests_enabled", secondState),
+						resource.TestCheckResourceAttr(resourceName, "pull_request_commenting_enabled", secondState),
+						resource.TestCheckResourceAttr(resourceName, "source_control_evaluation_enabled", secondState),
+						resource.TestCheckResourceAttr(resourceName, "base_branch", "my-cool-branch"),
 					),
 				),
 			},
@@ -80,8 +114,9 @@ func TestAccSourceControlOrganizationResource(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					resource.ComposeAggregateTestCheckFunc(
 						resource.TestCheckResourceAttr(resourceName, "remediation_pull_requests_enabled", firstState),
-						resource.TestCheckResourceAttr(resourceName, "remediation_pull_requests_enabled", firstState),
-						resource.TestCheckResourceAttr(resourceName, "remediation_pull_requests_enabled", firstState),
+						resource.TestCheckResourceAttr(resourceName, "pull_request_commenting_enabled", firstState),
+						resource.TestCheckResourceAttr(resourceName, "source_control_evaluation_enabled", firstState),
+						resource.TestCheckResourceAttr(resourceName, "base_branch", "my-cool-branch"),
 					),
 				),
 			},
@@ -100,8 +135,9 @@ func TestAccSourceControlOrganizationResource(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					resource.ComposeAggregateTestCheckFunc(
 						resource.TestCheckResourceAttr(resourceName, "remediation_pull_requests_enabled", secondState),
-						resource.TestCheckResourceAttr(resourceName, "remediation_pull_requests_enabled", secondState),
-						resource.TestCheckResourceAttr(resourceName, "remediation_pull_requests_enabled", secondState),
+						resource.TestCheckResourceAttr(resourceName, "pull_request_commenting_enabled", secondState),
+						resource.TestCheckResourceAttr(resourceName, "source_control_evaluation_enabled", secondState),
+						resource.TestCheckResourceAttr(resourceName, "base_branch", "my-cool-branch"),
 					),
 				),
 			},
@@ -130,6 +166,25 @@ resource "sonatypeiq_source_control" "test" {
   source_control_evaluation_enabled = %s
   repository_url = "https://github.com/sonatype-nexus-community/terraform-provider-sonatypeiq.git"
 }`, rand, rand, enabled, enabled, enabled)
+}
+
+func testAccSourceControlApplicationResourceMinimumConfig(rand string) string {
+	return fmt.Sprintf(providerConfig+`
+data "sonatypeiq_organization" "sandbox" {
+  name = "Sandbox Organization"
+}
+
+resource "sonatypeiq_application" "app_by_public_id" {
+  name = "app-%s"
+  public_id = "app-%s"
+  organization_id = data.sonatypeiq_organization.sandbox.id
+}
+
+resource "sonatypeiq_source_control" "test" {
+  owner_type = "application"
+  owner_id = sonatypeiq_application.app_by_public_id.id
+  repository_url = "https://github.com/sonatype-nexus-community/terraform-provider-sonatypeiq.git"
+}`, rand, rand)
 }
 
 func testAccSourceControlOrganizationResource(rand string, enabled string) string {

